pkg/debug/start_debug.go

```diff
@@ -0,0 +1,163 @@
+	deployment.Spec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(context, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(context, clusterNamespace, deployment.Spec)
+	if deployment_Pod_Name == "" || err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := getAndUpdatetheScale(context, clusterNamespace, deployment.Name, 0); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deployment_Pod_Name)
+
+	err = waitForPodDeletion(context, clusterNamespace, deployment_Pod_Name)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	debug_deployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deployment.Spec,
+	}
+
+	debugDeployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Create(ctx.TODO(), debug_deployment, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debug_deployment, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := getAndUpdatetheScale(context, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func getAndUpdatetheScale(context *k8sutil.Context, clusterNamespace, deploymentName string, scaleCount int) error {
+	scale, err := context.Clientset.AppsV1().Deployments(clusterNamespace).GetScale(ctx.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get scale of deployment %s. %v\n", deploymentName, err)
+	}
+
+	scale.Spec.Replicas = int32(scaleCount)
+	_, err = context.Clientset.AppsV1().Deployments(clusterNamespace).UpdateScale(ctx.TODO(), deploymentName, scale, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update scale of deployment %s. %v\n", deploymentName, err)
+	}
+	return nil
+}
+
+func verifyDeploymentExists(context *k8sutil.Context, clusterNamespace, deploymentName string) (*appsv1.Deployment, error) {
+	deployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Get(ctx.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
+}
+
+func updateDeployment(context *k8sutil.Context, clusterNamespace string, deploymentName *appsv1.Deployment) error {
+	_, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Update(ctx.TODO(), deploymentName, v1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func podRunningWithLabel(context *k8sutil.Context, clusterNamespace string, deploymentSpec appsv1.DeploymentSpec) (string, error) {
+
+	pods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), v1.ListOptions{})
```

Comment:
In the `ListOptions`, you can use the label selectors directly, so you will only get the pod needed instead of all pods in the namespace.

Reply:
here we don't know what deployment name the user will be passing, for example, it can be
`rook-ceph-mon-{a,b,c...} and rook-ceph-osd-{0,1,2,..}` so that's why I'm comparing the labels between the deployment passed and the pod we are getting

Reply:
Something like this should work:
```suggestion
	labelSelector := fmt.Sprintf("ceph_daemon_type=%s,ceph_daemon_id=%s", deploymentSpec.Template.Labels["ceph_daemon_type"], deploymentSpec.Template.Labels["ceph_daemon_id"])
	pods, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), v1.ListOptions{LabelSelector: labelSelector})
```

Reply:
thanks, it worked